kong/plugins/prometheus/exporter.lua

```diff
@@ -146,6 +150,21 @@ local function collect()
                  "/metrics endpoint: ", err)
   end
 
+  -- upstream targets accessible?
+  local balancer = require "kong.runloop.balancer"
+  local upstreams_dict = balancer.get_all_upstreams()
+  for name, id in pairs(upstreams_dict) do
+      local health_info = balancer.get_upstream_health(id)
+      for target, status in pairts(health_info) do
+          if status == "HEALTHY" then
+              metrics.upstream_target_health:set(1, { target })
+          else
```

Comment:
I was reading the documentation and it seems like this can be `DNS_ERROR` and `HEALTHCHECK_OFF`: https://docs.konghq.com/1.4.x/admin-api/#show-upstream-health-for-node

If that is true, we shouldn't be setting a target as unhealthy when health-checks are disabled. 

I'm not completely familiar with the internals here, could either of you, @locao or @hishamhm pitch in?

Reply:
Yes, that is true. You will be getting one of these four values: `HEALTHY`, `UNHEALTHY`, `DNS_ERROR` or `HEALTHCHECKS_OFF`.

I don't know how Prometheus stats are represented. If that metric is meant to add up to "number of healthy targets", then 1 on `HEALTHY` and 0 on anything else is correct. If you want to provide the actual health info for the individual target, then it's a enum of 4 possible values.

Reply:
Alright, so having a binary health metric is okay.
But, let's not export health of a target if `HEALTHCHECKS_OFF` is set.

Reply:
Incorporated.